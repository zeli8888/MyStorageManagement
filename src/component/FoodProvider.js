import React, { Component, useState, useEffect } from 'react';
import IngredientService from '../service/IngredientService'
import DishService from '../service/DishService'

export const FoodContext = React.createContext();  //exporting context object

const FoodProvider = ({ children }) => {
    const [ingredients, setIngredients] = useState([]);
    const [dishes, setDishes] = useState([]);

    useEffect(() => {
        IngredientService.getAllIngredients().then(response => { setIngredients(response.data) })
        DishService.getAllDishes().then(response => { setDishes(response.data) })
    }, []);

    return (
        <FoodContext.Provider value={
            {
                ingredients: ingredients,
                setIngredients: setIngredients,
                dishes: dishes,
                setDishes: setDishes
            }}>
            {children
                //this indicates that all the child tags with MyProvider as Parent can access the global store  
            }
        </FoodContext.Provider>)
}

export default FoodProvider;