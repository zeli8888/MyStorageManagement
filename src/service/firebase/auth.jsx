import {
    GoogleAuthProvider,
    GithubAuthProvider,
    OAuthProvider,
    signInWithPopup,
    setPersistence,
    browserSessionPersistence,
    signInWithEmailAndPassword,
    signOut,
    createUserWithEmailAndPassword,
    sendPasswordResetEmail,
} from 'firebase/auth';
import { firebaseAuth } from './firebaseConfig';

const googleProvider = new GoogleAuthProvider();
const githubProvider = new GithubAuthProvider();
const microsoftProvider = new OAuthProvider('microsoft.com');

microsoftProvider.setCustomParameters({
    prompt: 'select_account',
});

microsoftProvider.addScope('openid');
microsoftProvider.addScope('email');
microsoftProvider.addScope('profile');

// Sign in with Google functionality
export const signInWithGoogle = async () => {
    try {
        await setPersistence(firebaseAuth, browserSessionPersistence);
        const result = await signInWithPopup(firebaseAuth, googleProvider);
        return {
            success: true,
            user: result.user,
            error: null,
        };
    } catch (error) {
        return {
            success: false,
            user: null,
            error: error.message,
        };
    }
};

// Sign in with GitHub functionality
export const signInWithGithub = async () => {
    try {
        await setPersistence(firebaseAuth, browserSessionPersistence);
        const result = await signInWithPopup(firebaseAuth, githubProvider);
        return {
            success: true,
            user: result.user,
            error: null,
        };
    } catch (error) {
        return {
            success: false,
            user: null,
            error: error.message,
        };
    }
};

// Sign in with Microsoft functionality
export const signInWithMicrosoft = async () => {
    try {
        await setPersistence(firebaseAuth, browserSessionPersistence);
        const result = await signInWithPopup(firebaseAuth, microsoftProvider);
        return {
            success: true,
            user: result.user,
            error: null,
        };
    } catch (error) {
        return {
            success: false,
            user: null,
            error: error.message
        };
    }
};

// Sign in with email and password
export async function signInWithCredentials(email, password) {
    try {
        await setPersistence(firebaseAuth, browserSessionPersistence);
        const userCredential = await signInWithEmailAndPassword(
            firebaseAuth,
            email,
            password,
        );
        return {
            success: true,
            user: userCredential.user,
            error: null,
        };
    } catch (error) {
        return {
            success: false,
            user: null,
            error: error.message || 'Failed to sign in with email/password',
        };
    }
}

// Sign out functionality
export const firebaseSignOut = async () => {
    try {
        await signOut(firebaseAuth);
        return { success: true };
    } catch (error) {
        return {
            success: false,
            error: error.message,
        };
    }
};

// Sign up with email and password
export async function signUpWithCredentials(email, password) {
    try {
        const userCredential = await createUserWithEmailAndPassword(firebaseAuth, email, password);
        return { success: true, user: userCredential.user, error: null };
    } catch (error) {
        return { success: false, user: null, error: error instanceof Error ? error.message : "Registration failed" };
    }
}

// Send password reset email
export async function sendPasswordReset(email) {
    try {
        await sendPasswordResetEmail(firebaseAuth, email);
        return { success: true };
    } catch (error) {
        return { success: false, error: error instanceof Error ? error.message : "Password reset failed" };
    }
}

// Auth state observer
export const onAuthStateChanged = (callback) => {
    return firebaseAuth.onAuthStateChanged(callback);
};