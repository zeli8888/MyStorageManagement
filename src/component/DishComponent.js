import React, { useState, useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogContent from '@mui/material/DialogContent';
import DialogContentText from '@mui/material/DialogContentText';
import DialogTitle from '@mui/material/DialogTitle';
import DishService from '../service/DishService'
import { FoodContext } from './FoodProvider';
import PropTypes from 'prop-types';
import Box from '@mui/material/Box';
import Collapse from '@mui/material/Collapse';
import IconButton from '@mui/material/IconButton';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Typography from '@mui/material/Typography';
import Paper from '@mui/material/Paper';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
import MenuItem from '@mui/material/MenuItem';
import InputLabel from '@mui/material/InputLabel';
import ListItemText from '@mui/material/ListItemText';
import Select from '@mui/material/Select';
import Checkbox from '@mui/material/Checkbox';
import Grid from '@mui/material/Grid'
const DishComponent = function () {
    const navigate = useNavigate();
    const { ingredients, setIngredients, dishes, setDishes } = React.useContext(FoodContext);
    // const [dishes, setDishes] = useState([]);
    const [addingDish, setAddingDish] = useState(false);
    const [ingredientsForDish, setIngredientsForDish] = useState([]);

    useEffect(() => {
        refreshDishes();
    }, [navigate]);

    const refreshDishes = () => {
        DishService.getAllDishes().then(response => {
            setDishes(response.data);
        }).catch(error => {
            console.log(error);
        });
    }

    const updateDish = (data) => {
        DishService.updateDish(data.dishId, data).then(response => {
            refreshDishes();
        }).catch(error => {
            console.log(error);
        });
    }

    const addDish = (data) => {
        DishService.addDish(data).then(response => {
            setAddingDish(false);
            refreshDishes();
        }).catch(error => {
            console.log(error);
        });
    }

    const deleteDish = (dishId) => {
        DishService.deleteDish(dishId).then(response => {
            refreshDishes();
        }).catch(error => {
            console.log(error);
        });
    }

    function CreateDishRow(props) {
        const { dish } = props;
        const [open, setOpen] = React.useState(false);

        return (
            <React.Fragment>
                <TableRow sx={{ '& > *': { borderBottom: 'unset' } }}>
                    <TableCell>
                        <IconButton
                            aria-label="expand row"
                            size="small"
                            onClick={() => setOpen(!open)}
                        >
                            {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
                        </IconButton>
                    </TableCell>
                    <TableCell component="th" scope="row">
                        {dish.dishName}
                    </TableCell>
                    <TableCell>{dish.dishDesc}</TableCell>
                </TableRow>
                <TableRow>
                    <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={3}>
                        <Collapse in={open} timeout="auto" unmountOnExit>
                            <Box sx={{ margin: 1 }}>
                                <Typography variant="h6" gutterBottom component="div">
                                    Recipe
                                </Typography>
                                <Table size="small" aria-label="purchases">
                                    <TableHead>
                                        <TableRow>
                                            <TableCell>Name</TableCell>
                                            <TableCell align="right">Amount</TableCell>
                                            <TableCell align="right">Storage</TableCell>
                                            <TableCell align="right">Cost</TableCell>
                                            <TableCell>Description</TableCell>
                                        </TableRow>
                                    </TableHead>
                                    <TableBody>
                                        {dish.dishIngredients.map((dishIngredient) => (
                                            <TableRow key={dishIngredient.ingredient.ingredientId}>
                                                <TableCell component="th" scope="row">
                                                    {dishIngredient.ingredient.ingredientName}
                                                </TableCell>
                                                <TableCell align="right">{dishIngredient.dishIngredientQuantity}</TableCell>
                                                <TableCell align="right">{dishIngredient.ingredient.ingredientStorage}</TableCell>
                                                <TableCell align="right">{dishIngredient.ingredient.ingredientCost}</TableCell>
                                                <TableCell>{dishIngredient.ingredient.ingredientDesc}</TableCell>
                                            </TableRow>
                                        ))}
                                    </TableBody>
                                </Table>
                            </Box>
                        </Collapse>
                    </TableCell>
                </TableRow>
            </React.Fragment>
        );
    }

    const addIngredientForDish = (event) => {
        const {
            target: { value },
        } = event;
        setIngredientsForDish(
            // On autofill we get a stringified value.
            typeof value === 'string' ? value.split(',') : value,
        );
    }

    CreateDishRow.propTypes = {
        dish: PropTypes.shape({
            dishId: PropTypes.number.isRequired,
            dishName: PropTypes.string.isRequired,
            dishDesc: PropTypes.string.isRequired,
            dishIngredients: PropTypes.arrayOf(
                PropTypes.shape({
                    dishIngredientQuantity: PropTypes.number.isRequired,
                    ingredient: PropTypes.shape({
                        ingredientId: PropTypes.number.isRequired,
                        ingredientName: PropTypes.string.isRequired,
                        ingredientStorage: PropTypes.number.isRequired,
                        ingredientCost: PropTypes.number.isRequired,
                        ingredientDesc: PropTypes.string.isRequired,
                    })
                }),
            ).isRequired,
        }).isRequired,
    };

    return (
        <div className="container">
            <TableContainer component={Paper}>
                <Table aria-label="collapsible table">
                    <TableHead>
                        <TableRow>
                            <TableCell />
                            <TableCell>Dish</TableCell>
                            <TableCell>Description</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        {dishes.map((dish) => (
                            <CreateDishRow key={dish.dishId} dish={dish} />
                        ))}
                    </TableBody>
                </Table>
            </TableContainer>

            <Button variant="contained" color="success"
                onClick={() => {
                    setAddingDish(true);
                }}
            >
                New Dish
            </Button>

            <Dialog
                open={addingDish}
                onClose={() => { setAddingDish(false) }}
                maxWidth="xs"
                fullWidth={true}
                slotProps={{
                    paper: {
                        component: 'form',
                        onSubmit: (event) => {
                            event.preventDefault();
                            const formData = new FormData(event.currentTarget);
                            const formJson = Object.fromEntries(formData.entries());
                            addDish(formJson);
                        },
                    },
                }}
            >
                <DialogTitle>New Dish</DialogTitle>
                <DialogContent>

                    <TextField
                        autoFocus
                        required
                        margin="dense"
                        name="dishName"
                        label="Dish Name"
                        type="text"
                        fullWidth
                        variant="standard"
                    />
                    <TextField
                        margin="dense"
                        name="dishDesc"
                        label="Dish Description"
                        type="text"
                        fullWidth
                        variant="standard"
                    />
                    <Box mt={4} />
                    <InputLabel id="add-ingredient-for-dish">Recipe</InputLabel>
                    <Select
                        name="ingredientsForDish"
                        labelId="add-ingredient-for-dish"
                        multiple
                        value={ingredientsForDish}
                        onChange={addIngredientForDish}
                        renderValue={(selected) => selected.join(', ')}
                        MenuProps={{
                            PaperProps: { style: { maxHeight: '45%' } }
                        }}
                        style={{ width: '100%' }}
                    >
                        {ingredients.map((ingredient) => (
                            <MenuItem key={ingredient.ingredientId} value={ingredient.ingredientName}>
                                <Checkbox checked={ingredientsForDish.includes(ingredient.ingredientName)} />
                                <ListItemText primary={ingredient.ingredientName} />
                            </MenuItem>
                        ))}
                    </Select>
                    <Grid container spacing={2} >
                        {ingredientsForDish.map((ingredient) => {
                            return (
                                <TextField
                                    key={ingredient}
                                    name={ingredient}
                                    label={ingredient + " Amount"}
                                    value={1}
                                    variant="standard"
                                    color="success"
                                    margin="dense"
                                    type="number"
                                    required
                                />
                            );
                        })}
                    </Grid>
                </DialogContent>
                <DialogActions>
                    <Button onClick={() => { setAddingDish(false) }}>Cancel</Button>
                    <Button type="submit">Save</Button>
                </DialogActions>
            </Dialog>
        </div >
    )
}

export default DishComponent;