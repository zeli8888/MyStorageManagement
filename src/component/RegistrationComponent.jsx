import React, { useState } from 'react';
import { useNavigate, Link } from 'react-router-dom';
import { TextField, Button, Box, Typography } from '@mui/material';
import { signUpWithCredentials } from '../service/firebase/auth';
import SessionContext from './UserProvider';

function RegistrationComponent() {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState('');
  const { setSession } = React.useContext(SessionContext);
  const navigate = useNavigate();

  const handleSubmit = async (e) => {
    e.preventDefault();
    const result = await signUpWithCredentials(email, password);

    if (result?.success && result?.user) {
      setSession({
        user: {
          name: result.user.displayName || '',
          email: result.user.email || '',
          image: result.user.photoURL || '',
        }
      });
      navigate('/');
    } else {
      setError(result.error || 'Registration failed');
    }
  };

  return (
    <Box sx={{
      maxWidth: { xs: '100%', sm: 400 },
      px: { xs: 2, sm: 0 },
      mx: 'auto',
      mt: { xs: 2, sm: 4 },
    }}>
      <Typography variant="h5" gutterBottom>Sign Up</Typography>
      <form onSubmit={handleSubmit}>
        <TextField
          fullWidth
          label="Email"
          type="email"
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          margin="normal"
          required
        />
        <TextField
          fullWidth
          label="Password"
          type="password"
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          margin="normal"
          required
        />
        {error && <Typography color="error">{error}</Typography>}
        <Button
          type="submit"
          fullWidth
          variant="contained"
          sx={{
            mt: { xs: 1, sm: 2 },
            py: 1.5,
          }}
        >
          Create Account
        </Button>
        <Typography sx={{
          mt: { xs: 1, sm: 2 },
          fontSize: { xs: '0.875rem', sm: '1rem' }
        }}>
          Already have an account? <Link to="/login">Sign In</Link>
        </Typography>
      </form>
    </Box>
  );
}

export default RegistrationComponent;
