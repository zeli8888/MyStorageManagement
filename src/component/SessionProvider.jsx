import { createContext, useState, useEffect } from 'react';
import { onAuthStateChanged } from '../service/firebase/auth';

export const SessionContext = createContext();

const SessionProvider = ({ children }) => {
  const [session, setSession] = useState(null);
  const [loading, setLoading] = useState(true);
  const [rememberMe, setRememberMe] = useState(false);

  useEffect(() => {
    // Returns an `unsubscribe` function to be called during teardown
    const unsubscribe = onAuthStateChanged((user) => {
      if (user) {
        setSession({
          user: {
            name: user.displayName || '',
            email: user.email || '',
            image: user.photoURL || '',
          },
        });
      } else {
        setSession(null);
      }
      setLoading(false);
    });

    return () => unsubscribe();
  }, []);

  return (
    <SessionContext.Provider value={
      {
        session,
        loading,
        setSession,
        setLoading,
        rememberMe,
        setRememberMe
      }}>
      {children}
    </SessionContext.Provider>)
}

export default SessionProvider;
