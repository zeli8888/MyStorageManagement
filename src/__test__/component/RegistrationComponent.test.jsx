import { describe, expect, test, vi, beforeEach } from 'vitest';
import { render, screen, fireEvent, waitFor, cleanup } from '@testing-library/react';
import RegistrationComponent from '../../component/RegistrationComponent';
import { SessionContext } from '../../component/SessionProvider';
import { useNavigate } from 'react-router-dom';
import { act } from 'react';
import { signUpWithCredentials } from '../../service/firebase/auth';
import '@testing-library/jest-dom/vitest'

// Mock react-router-dom hooks
vi.mock('react-router-dom', () => ({
    useNavigate: vi.fn(),
    Link: ({ children, to }) => <a href={to}>{children}</a>,
}));

// Mock Firebase authentication service
vi.mock('../../service/firebase/auth', () => ({
    signUpWithCredentials: vi.fn(),
}));

// Mock session context values
const mockSessionContext = {
    setSession: vi.fn(),
    rememberMe: false,
    setRememberMe: vi.fn(),
};

// Mock navigate function
const mockNavigate = vi.fn();

describe('RegistrationComponent', () => {
    beforeEach(() => {
        vi.clearAllMocks();
        useNavigate.mockImplementation(() => mockNavigate);
    });
    afterEach(cleanup)

    // Snapshot test
    test('renders correctly and matches snapshot', () => {
        const { asFragment } = render(
            <SessionContext.Provider value={mockSessionContext}>
                <RegistrationComponent />
            </SessionContext.Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });

    // Test form submission with valid credentials
    test('handles successful registration', async () => {
        const mockUser = {
            displayName: 'Test User',
            email: 'test@example.com',
            photoURL: 'http://example.com/avatar.jpg',
        };

        signUpWithCredentials.mockResolvedValue({
            success: true,
            user: mockUser,
        });

        render(
            <SessionContext.Provider value={mockSessionContext}>
                <RegistrationComponent />
            </SessionContext.Provider>
        );

        act(() => {
            fireEvent.change(screen.getByLabelText(/Email/i), {
                target: { value: 'test@example.com' },
            });
            fireEvent.change(screen.getByLabelText(/Password/i), {
                target: { value: 'password123' },
            });
            fireEvent.click(screen.getByText(/Create Account/i));
        })

        await waitFor(() => {
            expect(mockSessionContext.setSession).toHaveBeenCalledWith({
                user: {
                    name: 'Test User',
                    email: 'test@example.com',
                    image: 'http://example.com/avatar.jpg',
                },
            });
            expect(mockNavigate).toHaveBeenCalledWith('/');
        });
    });

    // Test form submission with null credentials
    test('handles successful registration', async () => {
        const mockUser = {
            displayName: null,
            email: null,
            photoURL: null,
        };

        signUpWithCredentials.mockResolvedValue({
            success: true,
            user: mockUser,
        });

        render(
            <SessionContext.Provider value={mockSessionContext}>
                <RegistrationComponent />
            </SessionContext.Provider>
        );

        act(() => {
            fireEvent.change(screen.getByLabelText(/Email/i), {
                target: { value: 'test@example.com' },
            });
            fireEvent.change(screen.getByLabelText(/Password/i), {
                target: { value: 'password123' },
            });
            fireEvent.click(screen.getByText(/Create Account/i));
        })

        await waitFor(() => {
            expect(mockSessionContext.setSession).toHaveBeenCalledWith({
                user: {
                    name: '',
                    email: '',
                    image: '',
                },
            });
            expect(mockNavigate).toHaveBeenCalledWith('/');
        });
    });

    // Test form submission with error
    test('handles registration failure', async () => {
        const errorMessage = 'Email already in use';
        signUpWithCredentials.mockResolvedValue({
            success: false,
            error: errorMessage,
        });

        render(
            <SessionContext.Provider value={mockSessionContext}>
                <RegistrationComponent />
            </SessionContext.Provider>
        );

        act(() => {
            fireEvent.change(screen.getByLabelText(/Email/i), {
                target: { value: 'existing@example.com' },
            });
            fireEvent.change(screen.getByLabelText(/Password/i), {
                target: { value: 'password123' },
            });
            fireEvent.click(screen.getByText(/Create Account/i));
        })

        await waitFor(() => {
            expect(screen.getByText(errorMessage)).toBeInTheDocument();
        });
    });

    // Test form submission with missing error message
    test('handles registration failure', async () => {
        signUpWithCredentials.mockResolvedValue({
            success: false,
            error: null,
        });

        render(
            <SessionContext.Provider value={mockSessionContext}>
                <RegistrationComponent />
            </SessionContext.Provider>
        );

        act(() => {
            fireEvent.change(screen.getByLabelText(/Email/i), {
                target: { value: 'existing@example.com' },
            });
            fireEvent.change(screen.getByLabelText(/Password/i), {
                target: { value: 'password123' },
            });
            fireEvent.click(screen.getByText(/Create Account/i));
        })

        await waitFor(() => {
            expect(screen.getByText(/Registration failed/i)).toBeInTheDocument();
        });
    });

    // Test input field validation
    test('validates required fields', async () => {
        render(
            <SessionContext.Provider value={mockSessionContext}>
                <RegistrationComponent />
            </SessionContext.Provider>
        );

        act(() => {
            fireEvent.click(screen.getByText(/Create Account/i));
        })

        await waitFor(() => {
            expect(screen.getByLabelText(/Email/i)).toBeInvalid();
            expect(screen.getByLabelText(/Password/i)).toBeInvalid();
        });
    });

    // Test login link presence
    test('contains login link', () => {
        render(
            <SessionContext.Provider value={mockSessionContext}>
                <RegistrationComponent />
            </SessionContext.Provider>
        );

        expect(screen.getByText(/Already have an account/i)).toBeInTheDocument();
    });
});
