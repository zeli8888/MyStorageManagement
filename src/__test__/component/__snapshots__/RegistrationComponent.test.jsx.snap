// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RegistrationComponent > renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1uzsq72"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-1f9xtph-MuiTypography-root"
    >
      Sign Up
    </h5>
    <form>
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-16bevx5-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r0:"
          id=":r0:-label"
        >
          Email
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="email"
            class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            required=""
            type="email"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1n64csd-MuiNotchedOutlined-root"
            >
              <span>
                Email *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-16bevx5-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-19qnlrw-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for=":r1:"
          id=":r1:-label"
        >
          Password
          <span
            aria-hidden="true"
            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-1ljffdk-MuiFormLabel-asterisk"
          >
             *
          </span>
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            autocomplete="new-password"
            class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r1:"
            required=""
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1n64csd-MuiNotchedOutlined-root"
            >
              <span>
                Password *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-i7we6i-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Create Account
      </button>
      <div
        class="MuiBox-root css-imd3jm"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-14r958l-MuiFormControlLabel-root"
        >
          <span
            class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall css-reoaak-MuiButtonBase-root-MuiSwitchBase-root-MuiCheckbox-root"
          >
            <input
              class="PrivateSwitchBase-input css-12xagqm-MuiSwitchBase-root"
              data-indeterminate="false"
              name="remember"
              type="checkbox"
              value="true"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
              data-testid="CheckBoxOutlineBlankIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1c26xtj-MuiTypography-root"
          >
            Remember me
          </span>
        </label>
        <p
          class="MuiTypography-root MuiTypography-body2 css-1urkax6-MuiTypography-root"
        >
          Already have an account? 
          <a
            href="/login"
          >
            Sign In
          </a>
        </p>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
